### PR TITLE
connect to API

### PR DESCRIPTION
**Pull request by iamsjunaid**
### In this milestone of the project, I've
- [x] used the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as the backend
- [x] Fetched books from the Bookstore API after the component has rendered
- [x] Refactored add book and remove book features to persist your changes in the server

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,39 +1,107 @@
...
-    title: 'Cacking the Coding Interview',
-    author: 'Gayle Laakmann',
-  },
-];
+const getBooksURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/wUxXlMuiwuovrvJw2spJ/books';
+
+// get books from the API
+export const getBooks = createAsyncThunk('getBooks', async (thunkAPI) => {
+  try {
+    const response = await fetch(getBooksURL);
```
Comment (Shisui6) [inline src/redux/books/booksSlice.js]: Nice job fetching data from the API. However, I noticed you used the fetch API to make your requests. The project requirements state that you should use axios to make your requests. So please refactor all your HTTPS requests to use axios.🙏

image

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,39 +1,107 @@
...
-      const newBook = { ...action.payload, id: state.length + 1 };
-      state.push(newBook);
-    },
     removeBook: (state, action) => {
-      const bookIndex = state.findIndex((book) => book.id === action.payload);
-      if (bookIndex !== -1) {
-        state.splice(bookIndex, 1);
-      }
+      const bookId = action.payload;
+      /* eslint-disable-next-line no-param-reassign */
```
Comment (Shisui6) [inline src/redux/books/booksSlice.js]: I noticed a lot of eslint disable statements in your code. I know some linter errors can be frustrating but please try to solve all of them without the use of eslint disable statements. This way you will improve yourself as a developer and know how to adopt best coding practices. Please remove all eslint disable statements and try to solve the linter errors yourself🙏

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @iamsjunaid, 

# Requires Changes 🍀
You have done a great job so far!👏👏
You still need to work on a few issues before moving on to the next project, but you're almost there🪲!

## Highlights🎯
- Good job connecting to the API ✅
- Add and remove functions work ✅


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 🪶
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (iamsjunaid) [reply src/redux/books/booksSlice.js]: Thanks for the feedback @Shisui6 . I have replaced  `fetch` with `axios`.

File: src/redux/books/booksSlice.js (hunk 2 above)
Comment (iamsjunaid) [reply src/redux/books/booksSlice.js]: I have removed all the `eslint-disable` statements from the code.

**Review by binhussen — APPROVED** [review]
Hi @iamsjunaid , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Connected to API ✔️
- Used createAsyncThunk✔️
- Well-structured Code ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
Hi @binhussen, Thanks for approving the PR😊.

**Pull request merged**